components/camera/camera.go

```diff
@@ -315,13 +315,33 @@ func WrapWithReconfigurable(r interface{}, name resource.Name) (resource.Reconfi
 	if reconfigurable, ok := c.(*reconfigurableCamera); ok {
 		return reconfigurable, nil
 	}
+	reconfigurable := newReconfigurable(c, name)
+
+	if mon, ok := c.(LivenessMonitor); ok {
+		mon.Monitor(func() {
+			reconfigurable.reconfigure(c)
+		})
+	}
+
+	return reconfigurable, nil
+}
+
+func newReconfigurable(c Camera, name resource.Name) *reconfigurableCamera {
 	cancelCtx, cancel := context.WithCancel(context.Background())
 	return &reconfigurableCamera{
 		name:      name,
 		actual:    c,
 		cancelCtx: cancelCtx,
 		cancel:    cancel,
-	}, nil
+	}
+}
+
+// A LivenessMonitor is responsible for monitoring the liveness of a camera. An example
+// is connectivity. Since the model itself knows best about how to maintain this state,
+// the reconfigurable offers a safe way to notify if a state needs to be reset due
+// to some exceptional event (like a reconnect).
+type LivenessMonitor interface {
```

Comment:
I believe this solves the "how to resume streams properly on disconnect/reconnect" without having double use of a Reconfigurable -> Reconfigurable -> Camera

Reply:
later down the line, if we feel like this is a stable concept that could have value in other resources, it could be extracted out of this package. I don't want to have us make that call now because it may be better not using inversion of control and instead being a bit more opinionated about how monitoring works (say, via resource registration options).